### PR TITLE
[systemtest] Correct order of resource deletion in tests + update to Helm

### PR DESCRIPTION
**Pull request by im-konge**
### Type of change

- Bugfix

### Description

The issue with the order of resource deletion was found during Helm pipeline.
Resources were deleted not in chronological order - they were deleted in random order. That caused an issue when the Namespace, where particular Helm chart was installed, was deleted before the chart.
That affected all other tests (that are not using `infra-namespace`).

This PR should fix the order in which are the resources deleted. 
-----------------------------------------
Also, there was an issue with deletion of PVCs in KafkaNodePool mode -> when KafkaNodePools are enabled and used, we should delete PVCs and wait for their deletion during KafkaNodePool removal. Instead, we deleted the PVCs during Kafka resource deletion (which is before KafkaNodePool).

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Review by see-quick — APPROVED** [review]
LGTM, Assuming the tests will pass. Thanks! 👍

File: test/src/main/java/io/strimzi/test/k8s/HelmClient.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public HelmClient delete(String releaseName) {
      */
     public HelmClient delete(String namespace, String releaseName) {
         LOGGER.info("Deleting helm-chart:{} in namespace:{}", releaseName, namespace);
-        Exec.exec(null, command("delete", releaseName, "--namespace", namespace), 0, Level.DEBUG, false);
+        Exec.exec(null, command("uninstall", releaseName, "--namespace", namespace, "--wait"), 0, Level.DEBUG, true);
```
[the remark this answers is not in the document]
Comment (im-konge) [reply test/src/main/java/io/strimzi/test/k8s/HelmClient.java:71]:
```
> helm uninstall --help

Use the '--dry-run' flag to see which releases will be uninstalled without actually
uninstalling them.

Usage:
  helm uninstall RELEASE_NAME [...] [flags]

Aliases:
  uninstall, del, delete, un
```
it's an alias, so it really doesn't matter.
I just wanted to use option that is documented in Helm's documentation.

**Pull request merged**